### ISSUE TITLE
Logging level error panic

### ISSUE THREAD

**Issue by wutz** [iand/logfmtr#3]
```go
package main

import (
	"github.com/iand/logfmtr"
)

func main() {
	logfmtr.SetVerbosity(1)
	opts := logfmtr.DefaultOptions()
	base := logfmtr.NewWithOptions(opts)

	log := base.WithValues("user", "you")
	//log.Info("hello", "val1", 1, "val2", map[string]int{"k": 1})
	log.Error(nil, "uh oh", "trouble", true, "reasons", []float64{0.1, 0.11, 3.14})
}
```

Logging level error first will be panic.

It will be work correctly if I add level info before error.

**Comment by wutz** [comment]
Panic info:

```go
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x58 pc=0x10a8bef]

goroutine 1 [running]:
github.com/iand/logfmtr.(*core).key(...)
	/tmp/logfmtr/logfmtr.go:333
github.com/iand/logfmtr.(*core).flatten(0x0, 0xc000114040, 0x4, 0x4, 0x40, 0xc000114040)
	/tmp/logfmtr/logfmtr.go:324 +0x14f
github.com/iand/logfmtr.(*sink).Error(0xc000100090, 0x0, 0x0, 0x10d01ac, 0x5, 0xc000114040, 0x4, 0x4)
	/tmp/logfmtr/logfmtr.go:175 +0x5c
github.com/go-logr/logr.Logger.Error(...)
	/Users/wutz/.go/path/pkg/mod/github.com/go-logr/logr@v1.0.0/logr.go:221
main.main()
	/tmp/logfmtr/examples/example.go:14 +0x36b
exit status 2
```

**Comment by iand** [comment]
Thanks for the report and the reproducer. Fixed in version 0.2.1
